crates/core/src/binding.rs

```diff
@@ -392,6 +385,7 @@ impl<'a> Binding<'a> {
                         }
                         Some(Range {
                             start: Position::new(
+                                // FIXME: Should this be `leading_node`?
```

Comment:
I'd like to call out this FIXME, since I'm not sure if there's a bug or not. If it should indeed be `leading_node`, it would make it easy to extract a common helper so I can reduce duplication with the next method.

Reply:
I'm thinking it might be an option to drop `Binding::position()` altogether and rely on `CodeRange` within `ResolvedPattern` only.

In `regex.rs` I see a comment saying:
```
// this moves the byte-range out of sync with
// the row-col range, maybe we should just
// have a Range<usize> for String bindings?
```

I think it would simplify things, although the downside is that rows/columns may need to be calculated on-demand when we need them. So I'm not sure if it will pay off.

Reply:
what does leading_node refer to here? I don't see it anywhere in the code base. Calculating rows/columns on the fly is expensive, we do it somewhere, but best to be avoided if possible

Reply:
Sorry, I meant `leading_comment`. If you look at the constructed Range, `trailing_comment` is used for both the `Position` and `end_byte`. But for the starting byte there’s a difference with the `Position`.

Reply:
looks like a bug to me since, the byte range and the row/col mismatch

Reply:
should also be using the is_comment utility in language/language.rs